Common/Core/RecoDecay.h

```diff
@@ -285,6 +286,18 @@ struct RecoDecay {
     return static_cast<double>(length) * static_cast<double>(mass) / p(mom);
   }
 
+  /// Calculates pseudoproper decay length.
+  /// \param posPV  {x, y, z} or {x, y} position of the primary vertex
+  /// \param posSV  {x, y, z} or {x, y} position of the secondary vertex
+  /// \param mom  {x, y, z} or {x, y} momentum array
+  /// \return pseudoproper decay length
+  template <std::size_t N, typename T, typename U, typename V, typename M = double>
+  static double pseudoProprDL(const T& posPV, const U& posSV, const std::array<std::array<V, 3>, N>& mom, const std::array<M, N> mass)
+  {
+    // t_xy = l_xy * m / pT
+    return distanceXY(posPV, posSV) * m(mom, mass) / std::apply([](const auto&... args) { return pt(args...); }, mom);
+  }
+
```

Comment:
Why can you not use `ct`?

Reply:
It's a slightly different variable, as with the pseudoproper decay length we can use the better resolution in the xy plane. It was also used by several papers so I would like to add it to our analysis as well

Reply:
Ok, I renamed it, thank you!